llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1750,9 +1751,16 @@ SmallVector<int64_t> computeInterchangeFromDimPos(ArrayRef<int64_t> dimsPos,
 }
 
 // Implements `getIterationDomain` from the tiling interface.
+// TODO(hanchung): Add support for pad=true.
+// We might want to revisit the implementation. Taking a 2D input tensor as an
+// example, we might want loops ([0, ceilDiv(N, n), 1], [0, ceilDiv(M, m), 1]).
+// The scalar implementation generates two scf.for loops ([0, n, 1], (0, m, 1]).
+// For padding, we shoulc check if it's in bound. If so, load from the source.
+// If not, yield zero instead.
```

Comment:
@chelini @MaheshRavishankar @nicolasvasilache @rengolin I have a prototype on adding pad semantics. I feel that we'd like to update iteration_types and generateScalarImplementation to what Mahesh suggested. It would make us easier on pad support and tiling interface support. I put the idea in the comment, please take a look.

Reply:
I made the change in #10569